api/python/t4/packages.py

```diff
@@ -217,9 +219,18 @@ def fetch(self, dest=None):
         Returns:
             None
         """
-        if dest is None:
-            name = pathlib.Path(_to_singleton(self.physical_keys)).name
-            name = name.split('?versionId=')[0]
+        # Generate name for the file regardless of received destination
+        name = pathlib.Path(_to_singleton(self.physical_keys)).name
+        name = name.split('?versionId=')[0]
+
+        # If provided a destination, resolve the path and determine directory or destination behavior
+        if dest:
+            dest = pathlib.Path(dest).expanduser().resolve()
+            # We can determine if filepath was passed or directory was passed based off suffix
+            # We can't use the `is_dir` or `is_file` methods because the directory or file may not exist
+            if dest.suffix == "":
+                dest /= name
```

Comment:
It took me a couple seconds to understand what this is doing. This is probably _too_ clever, it's more readable to just do `dest = name / dest`.

Reply:
Fixed. Force pushed because I didn't want a commit for a single line change. 🤷‍♂

Reply:
We squash commits when we merge anyway.